test/SparqlExpressionTest.cpp

```diff
@@ -807,16 +807,26 @@ TEST(SparqlExpression, isSomethingFunctions) {
 
 // ____________________________________________________________________________
 TEST(SparqlExpression, testToNumericExpression) {
-  auto checkGetInt = testUnaryExpression<&makeIntExpression>;
-  auto checkGetDouble = testUnaryExpression<&makeDoubleExpression>;
-
-  checkGetInt(idOrLitOrStringVec({U, "  -1275", "5.97", "-78.97", "-5BoB6",
-                                  "FreBurg1", "", " .", " 42\n", " 0.01 ", ""}),
-              Ids{U, I(-1275), U, U, U, U, U, U, I(42), U, U});
-  checkGetDouble(
+  auto checkStrGetInt = testUnaryExpression<&makeIntExpression>;
+  auto checkStrGetDouble = testUnaryExpression<&makeDoubleExpression>;
+  auto checkNumGetInt = testUnaryExpression<&makeIntExpression>;
+  auto checkNumGetDouble = testUnaryExpression<&makeDoubleExpression>;
+
+  checkStrGetInt(
+      idOrLitOrStringVec({U, "  -1275", "5.97", "-78.97", "-5BoB6", "FreBurg1",
+                          "", " .", " 42\n", " 0.01 ", ""}),
+      Ids{U, I(-1275), U, U, U, U, U, U, I(42), U, U});
+  checkStrGetDouble(
       idOrLitOrStringVec({U, "-122.2", "19,96", " 128789334.345 ", "-0.f",
                           "  0.007 ", " -14.75 "}),
       Ids{U, D(-122.2), U, D(128789334.345), U, D(0.007), D(-14.75)});
+  checkNumGetInt(idOrLitOrStringVec({U, I(-12475), I(42), I(0), D(-14.57),
+                                     D(33.0), D(0.00001)}),
+                 Ids{U, I(-12475), I(42), I(0), I(-14), I(33), I(0)});
+  checkNumGetDouble(
+      idOrLitOrStringVec(
+          {U, I(-12475), I(42), I(0), D(-14.57), D(33.0), D(0.00001)}),
+      Ids{U, D(-12475.00), D(42.00), D(0.00), D(-14.57), D(33.00), D(0.00001)});
```

Comment:
I think we currently (for reasons of laziness of other people in previous times and because it used to be harder than now) never test the case that we convert VocabIds or LocalVocabIds using the expressions.
This could be a nice side project for you for a follow-up PR . You "just" have to take control over the knowledge graph that is implicitly built by all the tests here, s.t. you know which elements are contained.
But this is something we can talk about at another time, if it is worth your time.

Reply:
Seems like i would learn a lot about sparqlExpressions and the graph structure itself, would try it.